metricbeat/module/munin/node/node.go

```diff
@@ -42,53 +42,63 @@ func init() {
 // interface methods except for Fetch.
 type MetricSet struct {
 	mb.BaseMetricSet
-	namespace string
-	timeout   time.Duration
+	serviceName string
+	serviceType string
+	timeout     time.Duration
 }
 
 // New creates a new instance of the MetricSet. New is responsible for unpacking
 // any MetricSet specific configuration options if there are any.
 func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	cfgwarn.Beta("The munin node metricset is beta.")
 
-	config := struct {
-		Namespace string `config:"node.namespace" validate:"required"`
-	}{}
+	config := defaultConfig
 	if err := base.Module().UnpackConfig(&config); err != nil {
 		return nil, err
 	}
 
 	return &MetricSet{
 		BaseMetricSet: base,
-		namespace:     config.Namespace,
+		serviceName:   config.ServiceName,
+		serviceType:   config.ServiceType,
 		timeout:       base.Module().Config().Timeout,
 	}, nil
 }
 
 // Fetch method implements the data gathering
-func (m *MetricSet) Fetch() (common.MapStr, error) {
+func (m *MetricSet) Fetch(r mb.ReporterV2) {
 	node, err := munin.Connect(m.Host(), m.timeout)
 	if err != nil {
-		return nil, err
+		r.Error(err)
+		return
 	}
 	defer node.Close()
 
 	items, err := node.List()
 	if err != nil {
-		return nil, err
+		r.Error(err)
+		return
 	}
 
-	event, err := node.Fetch(items...)
+	metrics, err := node.Fetch(items...)
 	if err != nil {
-		return nil, err
+		r.Error(err)
 	}
 
-	// Set dynamic namespace.
-	_, err = event.Put(mb.NamespaceKey, m.namespace)
-	if err != nil {
-		return nil, err
+	// Even if there was some error, keep sending succesfully collected metrics if any
+	if len(metrics) == 0 {
+		return
 	}
-
-	return event, nil
-
+	event := mb.Event{
+		Service: m.serviceType,
```

Comment:
The service type should not be `munin` by the service that is monitored. So if `apache` is monitored with munin, `service.type` should be apache. Because of this we should also have it set by the user. I wonder if we should make it required for the user to have it set or make it optional?

Reply:
> I had a discussion with @exekias and he proposed to have this as general option for all modules instead. Can you remove it from here and potentially open a separate PR with introducing these settings?

I totally agree with having an option for `service.name`. Not so sure for `service.type`, I think that the type should be set in general by the module (do we want to allow arbitrary `service.type` for service modules?)

Reply:
* +1 on only allowing to set the `service.type` for the "input" modules. We should not allow to have it overwritten in other modules
* It's very interesting that you set the observer above as munin and not metricbeat. @webmat This gets even more interesting now ;-)
* I like the idea of the plugin field. How easily can this mapping be done? I assume we only have data from 1 plugin at the time?

Reply:
I will add a general option for `service.name` in other PR.

The setting of `service.type` and `observer.type` will also go in another PR, I think there can be some shared code for that in all "input" modules.

The plugin field would be quite easy, the loop on `items` [here](https://github.com/elastic/beats/pull/10322/files#diff-c25214dbd1610050e6ba92aff2c8903fR93), should rather be called `plugins` :slightly_smiling_face:. I'll give a try to this here.

Reply:
PR to add `service.name` https://github.com/elastic/beats/pull/10427